lua/telescope/config.lua

```diff
@@ -15,13 +16,36 @@ local function first_non_null(...)
   end
 end
 
-local sorters = require('telescope.sorters')
+-- A function that creates an amended copy of the `base` table,
+-- by replacing keys at "level 2" that match keys in "level 1" in `priority`,
+-- and then performs a deep_extend.
+-- May give unexpected results if used with tables of "depth"
+-- greater than 2.
+local smarter_depth_2_extend = function(priority, base)
+  local result = {}
+  for key, val in pairs(base) do
+    if type(val) ~= "table" then
+      result[key] = first_non_null(priority[key], val)
+    else
+      result[key] = {}
+      for k, v in pairs(val) do
+        result[key][k] = first_non_null(priority[k], v)
+      end
+    end
+  end
+  result = vim.tbl_deep_extend("keep", priority, result)
+  return result
+end
+
+local sorters = require "telescope.sorters"
 
 -- TODO: Add other major configuration points here.
 -- selection_strategy
 
 local config = {}
 
+config.smarter_depth_2_extend = smarter_depth_2_extend
```

Comment:
do we need this? I don't see the function used outside of `config` but if we do at some point we should put it in utils instead

Reply:
I think this was a leftover from my initial attempts at setting up tests in `layout_strategies_spec.lua`.
It can be safely removed.